exportoptionsgenerator/exportoptionsgenerator.go

```diff
@@ -21,128 +21,118 @@ const (
 	logger              log.Logger
 }
 
 // New constructs a new ExportOptionsGenerator.
 func New(xcodeProj *xcodeproj.XcodeProj, scheme *xcscheme.Scheme, configuration string, xcodeVersionReader xcodeversion.Reader, logger log.Logger) ExportOptionsGenerator {
-	g := ExportOptionsGenerator{
-		xcodeProj:          xcodeProj,
-		scheme:             scheme,
-		configuration:      configuration,
-		xcodeVersionReader: xcodeVersionReader,
-	}
-	g.certificateProvider = LocalCodesignIdentityProvider{}
-	g.profileProvider = LocalProvisioningProfileProvider{}
-	g.targetInfoProvider = XcodebuildTargetInfoProvider{xcodeProj: xcodeProj}
-	g.logger = logger
-	return g
+	targetInfoProvider := NewXcodebuildTargetInfoProvider(xcodeProj, scheme, configuration)
+	return NewWithInfoProvider(targetInfoProvider, xcodeVersionReader, logger)
+}
+
+// NewWithIosArchive constructs a new ExportOptionsGenerator with an xcarchive.
+func NewWithIosArchive(archive xcarchive.IosArchive, productToDistribute ExportProduct, xcodeVersionReader xcodeversion.Reader, logger log.Logger) ExportOptionsGenerator {
+	targetInfoProvider := NewIosArchiveInfoProvider(archive, productToDistribute)
+	return NewWithInfoProvider(targetInfoProvider, xcodeVersionReader, logger)
+}
+
+// NewWithInfoProvider constructs a new ExportOptionsGenerator with a custom InfoProvider.
+func NewWithInfoProvider(infoProvider InfoProvider, xcodeVersionReader xcodeversion.Reader, logger log.Logger) ExportOptionsGenerator {
+	return ExportOptionsGenerator{
+		xcodeVersionReader:  xcodeVersionReader,
+		certificateProvider: LocalCodesignIdentityProvider{},
+		profileProvider:     LocalProvisioningProfileProvider{},
+		archiveInfoProvider: infoProvider,
+		logger:              logger,
+	}
 }
 
 // GenerateApplicationExportOptions generates exportOptions for an application export.
-func (g ExportOptionsGenerator) GenerateApplicationExportOptions(
-	exportMethod exportoptions.Method,
-	containerEnvironment string,
-	teamID string,
-	uploadBitcode bool,
-	compileBitcode bool,
-	archivedWithXcodeManagedProfiles bool,
-	codeSigningStyle exportoptions.SigningStyle,
-	testFlightInternalTestingOnly bool,
-) (exportoptions.ExportOptions, error) {
+func (g ExportOptionsGenerator) GenerateApplicationExportOptions(exportMethod exportoptions.Method, codeSigningStyle exportoptions.SigningStyle, opts Opts) (exportoptions.ExportOptions, error) {
 	xcodeVersion, err := g.xcodeVersionReader.GetVersion()
 	if err != nil {
 		return nil, fmt.Errorf("failed to get Xcode version: %w", err)
 	}
 
-	mainTargetBundleID, entitlementsByBundleID, err := g.applicationTargetsAndEntitlements(exportMethod)
+	archiveInfo, err := g.archiveInfoProvider.Read()
 	if err != nil {
 		return nil, err
 	}
 
-	iCloudContainerEnvironment, err := determineIcloudContainerEnvironment(containerEnvironment, entitlementsByBundleID, exportMethod, xcodeVersion.Major)
+	// app-store-connect exports should contain App Clip too, but not other export methods:
```

Comment:
I would be great to clarify this comment a bit / write a new comment to document the relationship between the distributionBundleID and the provisioningProfiles fields:
- distribution bundle ID describes which app should be exported (main app / app-clip, only valid for none app-store export)
- when distributing an app-clip, its bundle ID needs to be in the  provisioningProfiles dictionary, otherwise it needs to be removed

Reply:
Updated comment.